### ISSUE TITLE
Resolution/Hz not returning to normal after stopping HDR playback

### ISSUE THREAD

**Issue by p1xel8ted** [xbmc/xbmc#21314]
### Describe the bug
Here is a clear and concise description of what the problem is:

Desktop res and UI set to 1080p/60. 
Fullscreen window
Adjust display refresh rate is turned on. 
If I play an HDR file it switches to the appropriate res (i.e 3840x2160) and refresh rate (i.e. 23.976) but doesn't switch back (to 1080p/60) on stop. Only does this with HDR content. Requires a Kodi restart to resolve.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

Switch back to the desktop resolution of 1080p/60. It used to but now doesn't for whatever reason. 

## Actual Behavior

It stays at whatever resolution and refresh rate the media was in. If it was a 4K movie, HDR switches off, but stays at 2160p/24hz. If it was a 1080p HDR file, HDR switches off, and stays at 24hz (as res is already 1080p)

### To Reproduce

1.  Ensure "Adjust display refresh rate" is on
2.  Ensure "Used display HDR capabilities" is on
3. Set UI and Desktop rez to 1080p/60hz/full-screen window
4. Play 4k/1080p HDR file and stop a few seconds after res and refresh has switched
5. 

### Debuglog

The debuglog can be found here:

[4K HDR Debug Log](https://paste.kodi.tv/utelepojek.kodi)
[1080p HDR Debug Log](https://paste.kodi.tv/raludihiri.kodi)

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

**Playing regular files, the following lines appear after stopping playback:**

DEBUG <general>: CWinSystemWin32::SetFullScreen: (windowed fullscreen) with size 1920x1080, refresh 60.000000
...
...
DEBUG <general>: CWinSystemWin32DX::FixRefreshRateIfNecessary: refreshRate: 60.0000, desired: 60.0000, deviation: 0.00000, fixRequired: no, 1

**This appears after stopping playback in all instances of HDR files, but there is no SetFullScreen called, which I assume is incorrect, which also results in the below (FixRefreshRateIfNecessary assuming nothing is wrong?)**

DEBUG <general>: CWinSystemWin32DX::FixRefreshRateIfNecessary: refreshRate: 24.0000, desired: 24.0000, deviation: 0.00000, fixRequired: no, 1

### Your Environment
Used Operating system:
 - [ X ] Windows
 - [DxDiag](https://paste.kodi.tv/izawazukoc)
 - Latest Nvidia drivers 512.15, have tried 3 previous releases with DDU.
 - Toggled all 3 "Game optimization" features off, made no difference.
 - Leads me to suspect it's the current W11 build I'm on. No way to roll back without starting fresh.

 - Operating system version/name: Windows 11 22H2, kernel: Windows NT x86 64-bit version 10.0.22598.200
 - Kodi versions: 
           Kodi (19.4 (19.4.0) Git:20220421-3340543568
           Kodi (19.4 (19.4.0) Git:20220302-e12e66e019)
           Kodi (20.0-ALPHA1 (19.90.101) Git:20220420-3f3a3c9484)

**Comment by thexai** [comment]
> Desktop res and UI set to 1080p/60.

This statement does not correspond to what the log says:

`2022-04-24 12:06:00.813 T:19772    INFO <general>: Desktop Resolution: 3440x1440 32Bit at 120Hz`

My recommendation is set desktop resolution to 3840x2160 at 60 Hz from Windows display settings (if native resolution of TV is 4K).

**Comment by thexai** [comment]
`DX::DeviceResources::CreateDeviceResources: device is created on adapter 'NVIDIA GeForce RTX 3080 Ti' with D3D_FEATURE_LEVEL_12_1`

Seems RTX 3080 Ti is capable of 4K  ;-) 
So what reason is there to want 1080p GUI / desktop?

**Comment by a1rwulf** [comment]
@thexai if I read the report correctly, it would still sty at 4k@24 instead of switching back to 4k@60.

**Comment by p1xel8ted** [comment]
Dual screen. One monitor at 3440x1440 @ 120Hz and 4K TV at 1080p/60. The TV won't do 4K@60hz (in RGB), so If I set to 4K, im stuck using Kodi at 24fps.

But yes, what @a1rwulf said, regardless of what Kodi is set at, it doesn't switch back to the resolution and refresh rate prior to starting the HDR file

**Comment by p1xel8ted** [comment]
Ok, @thexai  - If I set Windows TV res (and Kodi) to 3840x2160, 60hz, 10bit, YCbCr420, Limited (only settings I can use at 60 to not get a black screen) - and I play a regular 1080p24 file, it doesn't switch anything - just plays. My TV will normally display "1080p/24" in the top left when it has switched to that mode, which it doesn't in the new configuration (because it's just playing everything in 2160p instead of switching.

It also means I can't play any file that is above 23/24h i.e. a lot of British stuff is 25 which bumps it up to 50hz - and the display goes blank.

So back to the original setup.

**Comment by thexai** [comment]
> Never had this issue prior, the only major change I've had is moving to W11, then moving to Beta Insider Branch about a month later - unfortunately, I can't go back without a full re-install to test if it's W11.

We cannot support "Windows 11 Beta Insider Branch", only stable Windows releases.

Plese note that Kodi works just fine in Windows 11 (final release) using Nvidia graphics with full HDR/4K capable TV (HDMI 2.0).

> 4K TV at 1080p/60. The TV won't do 4K@60hz (in RGB), so If I set to 4K, im stuck using Kodi at 24fps.

This TV is not best option for good 4k HDR experience in Windows...

All 4K HDR TVs must support 4K RGB 8bit 60Hz in SDR!!!!!  If not is a red flag which could even indicate that the HDMI is not 2.0 but 1.4

See: https://en.wikipedia.org/wiki/HDMI#Refresh_frequency_limits_for_standard_video

HDR in HDMI 1.4 is not supported.

I'm going to close this as Github issues are not for support, and in any case cannot accept an HDR related issue in "Windows 11 Beta insider" with a TV that not even supports 4K 60 Hz in SDR.

**Comment by thexai** [comment]
Use forum for support at forum.kodi.tv

**Comment by p1xel8ted** [comment]
> Use forum for support at forum.kodi.tv

I did - and got no reply. 

HDR works fine now, I went back to the current public release of W11 (22000.651) and it works as before. They added Flipmode stuff in the 22H2 so something has changed in W11 WDDM that is causing the issue.

**Comment by tom-shell** [comment]
W11 22H2 is widely released now. After upgrade I'm experiencing this issue on Kodi 19.4 and 19.4 nightly from 2022-10-02. After stopping HDR content, refresh rate doesn't switch back to 60Hz and stays on the changed refresh rate to match the content e.g. 24hz while Kodi is opened. Closing Kodi results in refresh rate switch back to 60hz. Before W11 22H2 everything was fine. Non-HDR content is not affected - stopping content results in refresh rate switch as it should.

**Comment by p1xel8ted** [comment]
> W11 22H2 is widely released now. After upgrade I'm experiencing this issue on Kodi 19.4 and 19.4 nightly from 2022-10-02. After stopping HDR content, refresh rate doesn't switch back to 60Hz and stays on the changed refresh rate to match the content e.g. 24hz while Kodi is opened. Closing Kodi results in refresh rate switch back to 60hz. Before W11 22H2 everything was fine. Non-HDR content is not affected - stopping content results in refresh rate switch as it should.

Yeah, issue returned post 22H2 installation. I don't have it on me as I'm on my phone, but the log says the requested mode (the pre-switch res) doesn't exist and bails.

Workaround for content was to add the 1080p and 2160p resolutions to the whitelist, but it doesn't solve the UI being stuck at 24hz
